specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -62,81 +79,128 @@ enum FileStatus {
 }
 
+@doc("Aggregation type.")
 enum AggregationType {
-  @doc("Average value") Average,
-  @doc("Total count") Count,
-  @doc("Aggregation will be average in this case") None,
-  @doc("Total sum") Total,
-  @doc("90th percentile") Percentile90,
-  @doc("95th percentile") Percentile95,
-  @doc("99th percentile") Percentile99,
+  @doc("Average value.") Average,
+  @doc("Total count.") Count,
+  @doc("Aggregation will be average in this case.") None,
+  @doc("Total sum.") Total,
+  @doc("90th percentile.") Percentile90,
+  @doc("95th percentile.") Percentile95,
+  @doc("99th percentile.") Percentile99,
 }
 
+@doc("Metric unit.")
 enum MetricUnit {
-  @doc("No unit specified") NotSpecified,
-  @doc("Percentage") Percent,
-  @doc("Value count") Count,
-  @doc("Seconds") Seconds,
+  @doc("No unit specified.") NotSpecified,
+  @doc("Percentage.") Percent,
+  @doc("Value count.") Count,
+  @doc("Seconds.") Seconds,
   @doc("Milliseconds") Milliseconds,
   @doc("Bytes") Bytes,
   @doc("Bytes per second") BytesPerSecond,
   @doc("Count per second") CountPerSecond,
 }
 
+@doc("Time Grain")
 enum TimeGrain {
-  @doc("5 seconds, available only if test run duration is less than 10 minutes")
+  @doc("5 seconds, available only if test run duration is less than 10 minutes.")
   PT5S,
-  @doc("10 seconds, available only if test run duration is less than 10 minutes")
+
+  @doc("10 seconds, available only if test run duration is less than 10 minutes.")
   PT10S,
   @doc("1 minute") PT1M,
-  @doc("5 minutes, available only if test run duration is greater than 1 minute")
+
+  @doc("5 minutes, available only if test run duration is greater than 1 minute.")
   PT5M,
-  @doc("1 hour, available only if test run duration is greater than 1 minute")
+
+  @doc("1 hour, available only if test run duration is greater than 1 minute.")
   PT1H,
 }
 
+@doc("Intervals")
 enum Interval {
-  @doc("5 seconds, available only if test run duration is less than 10 minutes")
+  @doc("5 seconds, available only if test run duration is less than 10 minutes.")
   PT5S,
-  @doc("10 seconds, available only if test run duration is less than 10 minutes")
+
+  @doc("10 seconds, available only if test run duration is less than 10 minutes.")
   PT10S,
   @doc("1 minute") PT1M,
-  @doc("5 minutes, available only if test run duration is greater than 1 minute")
+
+  @doc("5 minutes, available only if test run duration is greater than 1 minute.")
   PT5M,
-  @doc("1 hour, available only if test run duration is greater than 1 minute")
+
+  @doc("1 hour, available only if test run duration is greater than 1 minute.")
   PT1H,
 }
 
-@doc("Load test model")
+@added(APIVersions.v2023_04_01_preview)
+@doc("Test type")
+enum TestType {
+  @doc("URL Test")
+  URL,
+
+  @doc("JMX Test")
+  JMX,
+}
+
+@doc("File info.")
+@resource("files")
+@parentResource(Test)
+model TestFileInfo extends FileInfo {
+  @doc("Name of the file.")
+  @key
+  @visibility
+  fileName: string;
+}
+
+@doc("File info.")
```

Comment:
nit: should the doc say test file info and test run file info

Reply:
Why do we need two different models for TestFileInfo and TestRunFileInfo?

Reply:
we have two paths tests/{testId}/files/{fileName} and test-run/{testRunId}/files/{fileName}, this means File is a child of both test and testrun. Therefore to facilitate this I had two models which extends same fileInfo but are child of test and testrun resources